src/pages/api/v1/applications/[id].ts

```diff
@@ -1,74 +1,88 @@
+    });
+  }
+};
+
+const updateHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  try {
+    if (typeof req.query.id === "string") {
+      const { id } = req.query;
+      const { status } = req.body;
+      const data = await prisma.application.update({
+        where: { id: parseInt(id) },
+        data: { status },
+      });
+      if (data) {
+        res.status(200).json(data);
       }
-    } catch (error) {
+    }
+  } catch (error) {
+    const { id } = req.query;
+    if (Number(id)) {
+      res.status(404).json({
+        message: `Could not find Application with id ${id}`,
+      });
+    } else {
       res.status(400).json({
         message: `Invalid Input: 'id' was not a number`,
       });
     }
-  } else if (req.method === 'PATCH') {
-    try {
-      if (typeof req.query.id === 'string') {
-        const { id } = req.query;
-        const { status } = req.body;
-        const data = await prisma.application.update({
-          where: { id: parseInt(id) },
-          data: { status },
-        });
-        if (data) {
-          res.status(200).json(data);
-        }
-      }
-    } catch (error) {
-      const { id } = req.query;
-      if (Number(id)) {
-        res.status(404).json({
-          message: `Could not find Application with id ${id}`,
-        });
-      } else {
-        res.status(400).json({
-          message: `Invalid Input: 'id' was not a number`,
-        });
+  }
+};
+
+const deleteHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const { id } = req.query;
+  try {
+    if (typeof id === "string") {
+      const data = await prisma.application.delete({
+        where: { id: parseInt(id) },
+      });
+      if (data) {
+        res.status(200).json(data);
       }
     }
-  } else if (req.method === 'DELETE') {
+  } catch (error) {
     const { id } = req.query;
-    try {
-      if (typeof id === 'string') {
-        const data = await prisma.application.delete({
-          where: { id: parseInt(id) },
-        });
-        if (data) {
-          res.status(200).json(data);
-        }
-      }
-    } catch (error) {
-      const { id } = req.query;
-      if (Number(id)) {
-        res.status(404).json({
-          message: `Could not find Application with id ${id}`,
-        });
-      } else {
-        res.status(400).json({
-          message: `Invalid Input: 'id' was not a number`,
-        });
-      }
+    if (Number(id)) {
+      res.status(404).json({
+        message: `Could not find Application with id ${id}`,
+      });
+    } else {
+      res.status(400).json({
+        message: `Invalid Input: 'id' was not a number`,
+      });
     }
   }
 };
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  if (req.method === "GET") {
+    getApplication(req, res);
+  } else if (req.method === "PATCH") {
+    updateHandler(req, res);
+  } else if (req.method === "DELETE") {
+    deleteHandler(req, res);
+  }
+};
```

Comment:
I've pushed a change to this file over three commits.

Please have a look and try to replicate this over to the other files.

Let me know if you have any questions.

Reply:
I try to "tell a story" with my commits. So if you like, you can step through my commits to see the process that I took.

Reply:
I have replicated this over to the other files.